TUI Implementation

Aarekaz opened the pull request:
This pull request introduces a new interactive TUI (Text User Interface) dashboard for managing and browsing local sessions, along with the necessary CLI command and supporting UI components. The dashboard provides a user-friendly way to view session statistics, browse sessions, filter/search, and navigate using keyboard shortcuts. The implementation includes new React components, hooks, and updates to documentation and dependencies.

**Major additions and changes:**

**1. Interactive Dashboard Feature**
- Added a new CLI command `sessionbase dashboard` to launch an interactive dashboard for session management, including keyboard navigation and platform filtering. (`README.md`, `src/commands/dashboard.ts`, `src/index.ts`) [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R42-R67) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R140-R142) [[3]](diffhunk://#diff-f3ef7602f2917e109bc12aff85ac53707705a0c5787b4ca62d392328e5ac770dR1-R19) [[4]](diffhunk://#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R28-R36)

**2. Dashboard UI Architecture**
- Implemented the main React app (`App.tsx`) for the dashboard, handling state, keyboard shortcuts, view switching (dashboard/sessions), search, filtering, and session detail navigation.
- Added a `MainContent` component to switch between the dashboard view and sessions view based on user input.
- Introduced a `Sidebar` and other supporting UI elements for navigation and context (not shown in the diff but referenced in `App.tsx`).

**3. Session Browsing and Search**
- Created a `SessionList` component for listing sessions with platform icons, previews, and details, supporting navigation and paging.
- Added a `SearchBar` component for searching/filtering sessions interactively.

**4. Dashboard Statistics and Visualization**
- Implemented an `ActivityHeatmap` component to visualize session activity over the last 28 days, showing active days and intensity.
- Added a reusable `BorderedBox` component for consistent UI framing.

**5. Dependency and Documentation Updates**
- Updated `package.json` to include new dependencies required for the TUI (e.g., `ink`, `react`, `date-fns`, and related types).
- Expanded the `README.md` with usage instructions, dashboard features, and keyboard shortcuts. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R42-R67) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R140-R142)

These changes collectively provide a robust, interactive way for users to manage and explore their sessions directly from the terminal.

Aarekaz committed:
initial design for a TUI

Aarekaz committed:
added initial details pane for sessions

Aarekaz committed:
Refine global quit shortcut to avoid conflicts in sessions view

Aarekaz committed:
Add detail expansion and paging functionality in sessions view

- Introduced state management for detail expansion and pagination in the App component.
- Updated the SessionsView to conditionally render either a preview or a full transcript based on the detailExpanded state.
- Implemented a new function to handle the rendering of the full transcript with pagination support.
- Enhanced the useSessionDetail hook to include fullLines for detailed session data.

Aarekaz committed:
added transcript view

Aarekaz committed:
added enter to search

Aarekaz committed:
desloped

owenmccadden commented:
<img width="641" height="360" alt="image" src="https://github.com/user-attachments/assets/d4476d37-fe4b-4baa-8aa7-072bbaeaec2e" />

This is very cool, wonder if we could make it more central / a bigger part of the dashboard?

Nit. I think the spacing might be a bit off here – should the `[1]` go left of "dashboard"?
<img width="171" height="260" alt="image" src="https://github.com/user-attachments/assets/5790e649-1507-4bc8-8495-117cf93f08cf" />

----

I think the `q` option to quit conflicts with the `q` option for Q chat if you go to the "sessions" menu

---

> Page 21 / 10 (200 lines) • use [[/] or PageUp/PageDown]

It looks like I can exceed the boundaries here, i.e. pagedown past page 10. Also not super clear to me what the ` use [/]` option means. I tried to use `/` but that triggers the search menu.

---

I wonder if we should show the full transcript in the sessions view of the dashboard, it feels a bit cluttered for long sessions. Maybe instead have a key shortcut to upload or view in the web UI?

Aarekaz committed:
Enhance README and TUI with new features and improvements

- Updated README to include detailed descriptions of the interactive TUI dashboard and its features.
- Added a new file to .gitignore for todo.md.
- Improved session listing in the TUI by adding date grouping for better organization.
- Enhanced the SessionsView layout for improved readability and navigation.
- Added keyboard shortcuts and usage hints for a better user experience.

Aarekaz committed:
Enhance ActivityHeatmap and DashboardView for improved user experience

- Updated ActivityHeatmap to use double-width blocks for better visual representation of activity levels.
- Adjusted layout in DashboardView to provide a more compact and informative header with inline stats.
- Added calculation for activity streak and average sessions per active day in DashboardView.
- Improved platform breakdown and recent sessions display for clearer insights.

Aarekaz committed:
Refactor TUI navigation and sidebar layout for improved usability

- Enhanced search focus handling in the App component to switch views more intuitively.
- Updated Sidebar component to include separators for better visual organization.
- Improved navigation menu layout and shortcut display for clearer user guidance.
- Compactified authentication status display for a cleaner look.

Aarekaz committed:
Refactor SearchBar component for improved focus handling

Aarekaz commented:
@owenmccadden, working on some design improvements. 

Dashboard:
<img width="1230" height="468" alt="image" src="https://github.com/user-attachments/assets/fbf696a4-63fb-4c16-a15b-5439f18b997b" />

<img width="1261" height="591" alt="image" src="https://github.com/user-attachments/assets/819d513e-cb9e-43fd-b282-603e7cdaae18" />


Session view:
<img width="1205" height="618" alt="image" src="https://github.com/user-attachments/assets/a46176f1-0fb2-4869-83aa-c410e9567252" />

Transcript: 

<img width="1220" height="636" alt="image" src="https://github.com/user-attachments/assets/0d20c0ea-61c0-4e7b-b54f-6bdaabff5834" />

I did fix the q issue too.

Aarekaz committed:
Update Sidebar and SessionsView components for layout adjustments

- Changed Sidebar component to use minWidth for better responsiveness.
- Reduced maxHeight in SessionsView's SessionList for a more compact display.

Aarekaz committed:
updated some mirnor issues and then the screenshot too for the readme

Aarekaz committed:
Refactor SessionList, Sidebar, and SessionsView components for improved organization and responsiveness

- Moved platform emoji and name functions to constants for better code reuse.
- Updated Sidebar component to dynamically calculate separator width for consistent layout.
- Replaced hardcoded platform data in SessionsView with a centralized PLATFORMS constant for easier maintenance.
- Removed redundant platform functions from SessionList and DashboardView for cleaner code.
